### PR TITLE
Extract IE version from IE cmake config

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [ ] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [ ] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

Based on https://github.com/ilya-lavrenov/openvino/pull/19 which is going to be merged after https://github.com/openvinotoolkit/openvino/pull/5484

CC @dkurt @alalek, please, comment before we commit changes to OpenVINO

### CODE REVIEW

File: cmake/OpenCVDetectInferenceEngine.cmake [hunk 1]
```diff
@@ -134,10 +134,25 @@ endif()
 # Add more features to the target
 
 if(INF_ENGINE_TARGET)
+  if(InferenceEngine_VERSION VERSION_GREATER_EQUAL "2021.4")
+    set(INF_ENGINE_RELEASE "${InferenceEngine_VERSION_MAJOR}")
```
Comment (mshabunin) [inline cmake/OpenCVDetectInferenceEngine.cmake:138]: Just curious, would `math` work here?
```suggestion
    math(EXPR INF_ENGINE_RELEASE "${InferenceEngine_VERSION_MAJOR} * 1000000 + ${InferenceEngine_VERSION_MINOR} * 10000 + ${InferenceEngine_VERSION_PATCH} * 100")
```

**Review by mshabunin — APPROVED** [review]

File: cmake/OpenCVDetectInferenceEngine.cmake [hunk 2]
```diff
@@ -134,10 +134,14 @@ endif()
 # Add more features to the target
 
 if(INF_ENGINE_TARGET)
+  if(InferenceEngine_VERSION VERSION_GREATER_EQUAL "2021.4")
+    math(EXPR INF_ENGINE_RELEASE "${InferenceEngine_VERSION_MAJOR} * 1000000 + ${InferenceEngine_VERSION_MINOR} * 10000 + ${InferenceEngine_VERSION_PATCH} * 100")
+  endif()
```
Comment (alalek) [inline cmake/OpenCVDetectInferenceEngine.cmake:139]: BTW, It makes sense to emit warning if there is passed wrong value through CMake's command-line (need to think how to ignore/handle "patch" version here in this comparison).

**Review by alalek — COMMENTED** [review]
Looks good to me :+1:

Please remove "draft" status when PR is ready.

**Review by alalek — APPROVED** [review]

**Comment by alalek** [comment]
> Please remove "draft" status when PR is ready.

@ilya-lavrenov Is this PR ready?

**Pull request merged**